From 46efb68e33f2401f6e472a05d19566a41b52bb99 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:32 -0400
Subject: [ethernet] bnx2x: Fix compilation when CONFIG_BNX2X_SRIOV is not set

Message-id: <1409946615-5862-28-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91571
O-Subject: [RHEL7.1 PATCH 27/70] bnx2x: Fix compilation when CONFIG_BNX2X_SRIOV is not set
Bugzilla: 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 19915f53af472e97d67a43b7b43f50af2830a1a8
Author: Yuval Mintz <Yuval.Mintz@qlogic.com>
Date:   Wed Mar 26 09:06:29 2014 +0200

    bnx2x: Fix compilation when CONFIG_BNX2X_SRIOV is not set

    Commit 370d4a26 "bnx2x: Create workqueue for IOV related tasks" breaks bnx2x
    compilation when CONFIG_BNX2X_SRIOV is not set - "multiple definition of
    `bnx2x_schedule_iov_task'".

    Reported-by: kbuild test robot <fengguang.wu@intel.com>
    Signed-off-by: Yuval Mintz <Yuval.Mintz@qlogic.com>
    Acked-by: Eric Dumazet <edumazet@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.h b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.h
index 2e3d6eb..96023f6 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.h
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_sriov.h
@@ -576,7 +576,7 @@ static inline int bnx2x_sriov_configure(struct pci_dev *dev, int num_vfs) {retur
 static inline void bnx2x_iov_channel_down(struct bnx2x *bp) {}
 
 static inline void bnx2x_iov_task(struct work_struct *work) {}
-void bnx2x_schedule_iov_task(struct bnx2x *bp, enum bnx2x_iov_flag flag) {}
+static inline void bnx2x_schedule_iov_task(struct bnx2x *bp, enum bnx2x_iov_flag flag) {}
 
 #endif /* CONFIG_BNX2X_SRIOV */
 #endif /* bnx2x_sriov.h */
-- 
1.7.1